### DISCUSSION TITLE
fyne serve fails: GopherJS 1.17.2+go1.17.9 requires a Go 1.17.x distribution, but found version go1.21.4

### DISCUSSION THREAD

**Discussion by prologic** [fyne-io/fyne#4504] [Q&A] [answered]
When running `fyne serve` and installing GopherJS as requested, it fails with:

```
$ fyne serve
GopherJS 1.17.2+go1.17.9 requires a Go 1.17.x distribution, but found version go1.21.4
time 2023-11-01T20:46:39Z

GopherJS 1.17.2+go1.17.9 requires a Go 1.17.x distribution, but found version go1.21.4
time 2023-11-01T20:46:39Z


error building application: exit status 1
```

<img width="1242" alt="Screenshot 2024-01-02 at 11 57 45" src="https://github.com/fyne-io/fyne/assets/1290234/4ec3b890-4d43-42a9-b2a2-b4a615ca7cbf">

**Comment by andydotxyz** [comment, marked as answer]
Yes, this is a limitation of the GopherJS build tool and we have documented how you can set it up at https://developer.fyne.io/started/webapp

**Reply by prologic** [reply]
Can we update the output messaging to reflect the right thing to do? 🤔

**Reply by andydotxyz** [reply]
The output is created by the gopherJS command.
But as @Jacalz said we decided to remove GopherJS for now, so in the next major release this will be gone anyway.

**Comment by Jacalz** [comment]
You can use `-target=wasm` to not build with GopherJS. It is also worth noting that GopherJS support will be removed in Fyne v2.5.0.
